c/misra/src/rules/RULE-2-5/UnusedMacroDeclaration.ql

```diff
@@ -19,7 +19,18 @@ import codingstandards.c.misra
 from Macro m
 where
   not isExcluded(m, DeadCodePackage::unusedMacroDeclarationQuery()) and
+  // We consider a macro "used" if there is a macro access
   not exists(MacroAccess ma | ma.getMacro() = m) and
+  // Or if there exists a check whether the macro is defined which the extractor
+  // hasn't been able to tie to a macro (usually because this use came before
+  // the macro was defined e.g. a header guard)
+  not exists(PreprocessorBranchDirective bd |
+    // Covers the #ifdef and #ifndef cases
+    bd.getHead() = m.getName()
+    or
+    // Covers the use of defined() to check if a macro is defined
+    m.getName() = bd.getHead().regexpCapture(".*defined\\((.*)\\).*", 1)
```

Comment:
I think it's worth making a small change here to better handle whitespace and missing parenthesis.

Some example test-cases, which should all have equivalent functionality:
`#if defined(VAR)`

`#if defined ( VAR )`

`#if defined VAR`

Reply:
Done - this required disabling clang-format for this section.